website/blog/2020-11-10-lithium.md

```diff
@@ -0,0 +1,507 @@
+
+Some more work was required in order to make the new features work smoothly with
+Metals and currently, thanks to [ckipp01](https://github.com/ckipp01), users can
+easily try out the new sbt BSP support. It's important to note that using Bloop
+instead of sbt is still the recommended approach as we believe it provides the
+best user experience and features like running or debugging are still not yet
+implemented for sbt. More details and explanations can be found in the
+[blogpost](LINK).
+
+## Remote debugging
+
+Thanks to the great work of [pvid](https://github.com/pvid), Metals now supports
+remote debugging, which means that it can attach to a running process with the
+proper JVM options specified. There is no longer a need to run the application
+or test from within the editor.
+
+In case of a simple java process those options will take a form of for example:
+
+`-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005`
+
+Some of the more important options here are:
+
+- `suspend=y` will make the process wait for Metals to connect to it if
+  specified to `y`. `n` will cause the process to run normally.
+- `address=5005` specifies which port to use and it can be any free port.
+
+For a detailed explanation of the different options please refer to the proper
+documentation
+[here](https://docs.oracle.com/javase/8/docs/technotes/guides/jpda/conninv.html)
+
+When using sbt, remote debugging can be achieved by specifying additional
+settings:
+
+```scala
+  javaOptions in run := List("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005"),
+  fork in run := true
+```
+
+This will cause running `sbt run` to wait for Metals to connect to it, which
+might be especially useful when reading user input, which is currently
+impossible to do for example from within VS Code. Similarly, these options can
+be specified in any of the supported build tools.
+
+To later connect to the running process, you need to use the additional Attach
+request type with the `buildTarget`, `hostname` and `port` fields specified. In
+case of Visual Studio Code this will take a form of:
+
+```json
+{
+  "type": "scala",
+  "request": "attach",
+  "name": "Attach",
+  "buildTarget": "root",
+  "hostName": "localhost",
+  "port": 5005
+}
+```
+
+## Environment variables in run/debug
+
+Metals now supports specifying additional environment options when running or
+debugging applications. This can be done twofold:
+
+- By specifying the new `env` field:
+
+```json
+{
+  "type": "scala",
+  "name": "Debug Main",
+  "request": "launch",
+  "mainClass": "Main",
+  "args": ["boom", "boom"],
+  "env": { "VARIABLE1": " 123" }
+}
+```
+
+- By using the `envFile` field:
+
+```json
+{
+  "type": "scala",
+  "name": "Debug Main",
+  "request": "launch",
+  "mainClass": "Main",
+  "args": ["boom", "boom"],
+  "envFile": "local.env"
+}
+```
+
+Where the `local.env` file can take a form of:
+
+```
+MY.VARIABLE=1
+MY-OTHER-VARIABLE=2
```

Comment:
Hey Tomasz, dots and dashes aren't actually allowed as per the docs [here](https://github.com/scalameta/metals/blob/main/docs/editors/debug-adapter-protocol.md#L86-L88).
Could we replace `.` and `-` with `_` to be consistent with the example from the [docs](https://github.com/scalameta/metals/blob/main/docs/editors/debug-adapter-protocol.md#L68-L89)?

Reply:
Good catch! I have no idea where I saw the example.

Reply:
I will fix this together with the bsp link.

Reply:
And should be fine. Let me know if anything is off, I will update later.